packages/core/entity-list/src/modules/list/sagas.js

```diff
@@ -139,11 +137,17 @@ export function* prepareEndpointUrl(endpoint, searchEndpoint, hasUserChanges) {
 
 export function* countEntities() {
   const {entityName} = yield select(inputSelector)
-  const {endpoint, searchEndpoint} = yield select(listSelector)
+  const {formDefinition, searchListFormDefinition} = yield select(listSelector)
   const {showSelectedRecords, selection} = yield select(selectionSelector)
 
   const regardSelection = !showSelectedRecords
   const query = yield call(getBasicQuery, regardSelection)
+
+  const formDefinitionToUse =
```

Comment:
could we somehow use the `getFormDefinition` function in `forms.js` here as well. So that we have this logic with `hasUserChanges` only once?